controller/src/test/java/io/pravega/controller/store/stream/StreamMetadataStoreTest.java

```diff
@@ -919,5 +930,44 @@ public void sizeTest() throws Exception {
         store.addStreamCutToRetentionSet(scope, stream, streamCut5, null, executor).get();
         // endregion
     }
+
+    @Test
+    public void bigLongTest() {
+        BigLong counter = new BigLong(0, 1L);
```

Comment:
For a class with so many bitwise operations where things may turn out weirdly, I strongly recommend a more comprehensive set of tests. Thoroughly tests boundary cases and cases where you need to increment the MSBs, as well as invalid cases, where you pass in bad inputs.